File: dotnet/Multiplayer/Rooms/Realtime/SenderInfo.cs
namespace Michitai.Multiplayer.Rooms.Realtime
{
    /// <summary>
    /// Information about the sender of a realtime message.
    /// Contains minimal sender details for realtime communication.
    /// </summary>
    public struct SenderInfo
    {
        /// <summary>
        /// Whether the sender is the host.
        /// </summary>
        public bool is_host { get; set; }

        /// <summary>
        /// The player ID of the sender.
        /// </summary>
        public int game_player_id { get; set; }

        /// <summary>
        /// The name of the sender.
        /// </summary>
        public string player_name { get; set; }
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/realtime/SenderInfo.java
package com.michitai.multiplayer.rooms.realtime;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Information about the sender of a realtime message.
 * Contains minimal sender details for realtime communication.
 */
public class SenderInfo {
    @JsonProperty("is_host")
    private boolean isHost;

    @JsonProperty("game_player_id")
    private int gamePlayerId;

    @JsonProperty("player_name")
    private String playerName;

    public boolean isHost() {
        return isHost;
    }

    public void setHost(boolean host) {
        isHost = host;
    }

    public int getGamePlayerId() {
        return gamePlayerId;
    }

    public void setGamePlayerId(int gamePlayerId) {
        this.gamePlayerId = gamePlayerId;
    }

    public String getPlayerName() {
        return playerName;
    }

    public void setPlayerName(String playerName) {
        this.playerName = playerName;
    }
}
